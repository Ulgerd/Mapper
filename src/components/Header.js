import React from 'react';
import styled from 'styled-components'

const StyledHeader = styled.h1`
  display: flex;
  justify-content: center;
  margin: 0;
  padding: 1em;
  background-color: lightgray;
`

export function Header() {
  return (
    <StyledHeader>
      Mapper
    </StyledHeader>
  );
}

export default Header;
