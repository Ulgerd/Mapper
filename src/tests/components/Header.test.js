import React from 'react';
import { mount } from 'enzyme'
import { Header } from '../../components/Header.js';
import renderer from 'react-test-renderer';

describe('Header component tests',()=>{
    let wrapper

    beforeEach(()=>{
        wrapper = mount(<Header />)
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('contains h1', () => {
        expect(wrapper.find('h1')).toHaveLength(1)
    });

    it(`component's h1 have proper value`, () => {
      expect(wrapper.find('h1').text()).toEqual('Mapper');
    });

    it('renders correctly', () => {
      const tree = renderer
        .create(<Header />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});
