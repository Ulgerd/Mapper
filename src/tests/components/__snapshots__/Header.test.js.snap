// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests renders correctly 1`] = `
<h1
  className="sc-bdVaJa hDuIHb"
>
  Mapper
</h1>
`;
